### PR TITLE
Implement Presence #1

### PR DESCRIPTION
**Pull request by jonathanleang**
- this implementation needs review, looks like it's working but no test cases yet.
- List<JToken> to JObject bit is ugly..
*don’t merge just yet*

### CODE REVIEW

**Comment by Mazyod** [comment]
Looks promising! Will take a look.

**Comment by bspinner** [comment]
What's the status here?

**Comment by Mazyod** [comment]
This PR is missing tests, and I'm not using Presence feature myself ... If there is demand for this feature, I'll be inclined to give it a shot, otherwise PRs welcome.

**Comment by jonathanleang** [comment]
We are not using Presence here as well now, so if anyone like to finlish this up is welcome to. I feel like its a bit buggy

**Comment by jonathanleang** [comment]
We are gonna use Presence in a new project, I took an other look at my changes before and saw part of the code was was incomplete from my previous attempt. I think I fixed it now still missing test cases to be sure...

File: Phoenix/Presence.cs [hunk 1]
```diff
@@ -1,3 +1,9 @@
+using UnityEngine;
```
Comment (Mazyod) [inline Phoenix/Presence.cs:1]: I don't think we should import Unity in the main project, as it is meant to work with Unity as well as any C# project.

File: Phoenix/Presence.cs [hunk 2]
```diff
@@ -39,7 +45,296 @@ namespace Phoenix {
 	//     }
 	//     let onlineUsers = Presence.list(state, listBy)
 	//
+
 	public class Presence {
-		// TODO
+
+
+		public static JObject syncState(JObject currentState, JObject newState, Action<string, JToken, JToken> onJoin = null, Action<string, JToken, JToken> onLeave = null){
```
Comment (Mazyod) [inline Phoenix/Presence.cs:52]: We need to work on the consistent naming here.

File: Phoenix/Presence.cs [hunk 3]
```diff
@@ -39,7 +45,296 @@ namespace Phoenix {
...
+			JObject state = (JObject)JObject.Parse(currentState.ToString());
```
Comment (Mazyod) [inline Phoenix/Presence.cs:53]: This is serializing currentState to string, then back to JObject. Perhaps there is a better way to clone the object if needed.

**Comment by Mazyod** [comment]
Doing this in #20

**Pull request closed without merging**